pulsar-broker/src/main/java/org/apache/pulsar/broker/service/TopicListService.java

```diff
@@ -57,7 +57,7 @@ public TopicListWatcher(TopicListService topicListService, long id,
                                 Pattern topicsPattern, List<String> topics) {
```

Comment:
It seems that the pattern gets compiled twice. Shouldn't the `removeTopicDomainScheme` happen before `TopicListWatcher` is constructed?

Reply:
looks like org.apache.pulsar.broker.service.ServerCnx#handleCommandWatchTopicList would be the location.

Reply:
https://github.com/apache/pulsar/blob/4ac4f3c91bfdb7307c642fcbd35040015b5a8dd6/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L3156-L3173